hal/src/dmac/transfer.rs

```diff
@@ -571,6 +601,43 @@ where
         self.complete
     }
 
+    /// Checks and clears the block transfer complete interrupt flag
+    #[inline]
+    pub fn block_transfer_interrupt(&mut self) -> bool {
+        self.chan
+            .as_mut()
+            .check_and_clear_interrupts(InterruptFlags::new().with_tcmpl(true))
+            .tcmpl()
+    }
+
+    /// Modify a completed transfer with new `source` and `destination`, restart
+    ///
+    /// Returns a Result containing the source and destination from the
+    /// completed transfer.
+    pub fn recycle(&mut self, mut source: S, mut destination: D) -> Result<(S, D)> {
```

Comment:
Is it problematic if the new source and destination buffers are consumed in the event the method returns `Err`?